carbon-design-system/carbon
# feat(react): enable textarea to limit text input to set amount of words

Co-authored-by: SamChinellato <49278203+SamChinellato@users.noreply.github.com>
Co-authored-by: stevenpatrick009 <106097350+stevenpatrick009@users.noreply.github.com>

Closes #13788 

Added functionality to textArea component to limit the number of words the user can type. Also updated the useAnouncer to handle new wording by allowing it to either display "x characters left" or "x words left" based on args passed in.

#### Changelog

**New**

- textArea component has now word counter mode and can now limit the amount of words user types
- added more tests for the textArea component

**Changed**

- useAnouncer can now take the entityName and use that to display the "x <entity name> left"
- added tests for useAnnouncer



#### Testing / Reviewing

- run the storybook for react components
- navigate to the textArea component
- select the "WithWordCounter" textArea type
- type in few words and observe the counter increase
- try to type in more words than the limit and observe that user can not add more words
- try to paste in text with more words than the limit and observe that the text has been cut to match the word limit
